### PR TITLE
Fix bug when clipping geometry collections in vector tiles

### PR DESCRIPTION
**Pull request by iverase**
when clipping a geometry in vector tiles, we need to clone the tile envelope. The issue is that we are doing it at the beginning of the process so for geometry collections when we have more than one geometry, we using the same envelope for all geometries which can lead to corrupted geometries. 

This PR ensures that the tile envelope is cloned for each geometry when needed.

### CODE REVIEW

File: x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java [hunk 1]
```diff
@@ -222,14 +222,34 @@ public void testManyIntersectingGeometries() {
         assertThat(builder.getFeatures(GeoTileUtils.toBoundingBox(2 * x + 1, 2 * y, z + 1)), iterableWithSize(1));
         assertThat(builder.getFeatures(GeoTileUtils.toBoundingBox(2 * x, 2 * y + 1, z + 1)), iterableWithSize(1));
         assertThat(builder.getFeatures(GeoTileUtils.toBoundingBox(2 * x + 1, 2 * y + 1, z + 1)), iterableWithSize(1));
+        final GeometryCollection<Geometry> WithinCollection = new GeometryCollection<>(
```
Comment (craigtaverner) [inline x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java:225]: Variable name should start with lower case letter.

**Review by craigtaverner — APPROVED** [review]
Looks OK to me, but I'm not sure exactly why this helps. The envelope was copied already before. Is it that the intersected results are later mutated outside this function? In that case I agree it makes sense for multiple copies. Or could it be that the `envelope.intersection` method mutates the envelope, which would be scary!

File: x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java (hunk 1 above)
Comment (iverase) [reply x-pack/plugin/vector-tile/src/test/java/org/elasticsearch/xpack/vectortile/feature/FeatureFactoryTests.java]: fixed

**Comment by iverase** [comment]
>Or could it be that the envelope.intersection method mutates the envelope

The problem is that the intersection method copies the coordinates by reference in the clipped geometry. So if one of the coordinates of the final geometry comes from the envelope, it gets mutated when we transform it to a mvt geometry as we do it in place.

**Pull request merged**